### PR TITLE
[bare-expo] Add detox testing for android

### PR DESCRIPTION
**Pull request by Kudo**
# Why

Originally, detox testing is not enabled on Android.
Though we had `expotools android-native-unit-tests --type instrumented`, unfortunately,  the test cases right now only test on modules themselves but not bare-expo.
With this PR, we could be more confident from bare-expo or test-suites regressions. 

# How

Fix debox build on Andriod.

# Test Plan

Make these works
`yarn android:detox:test:debug`
`yarn android:detox:test:release`

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] Add Android detox to CI.
- [x] Fix broken test-suites.
- [x] Fix test broken for UI blocked by dev-client onboarding activity.

### CODE REVIEW

File: apps/bare-expo/package.json [hunk 1]
```diff
@@ -67,13 +67,13 @@
       },
       "android.emu.debug": {
         "binaryPath": "android/app/build/outputs/apk/debug/app-debug.apk",
-        "build": "cd android && ./gradlew assembleDebug assembleAndroidTest -DtestBuildType=debug && cd ..",
+        "build": "cd android && ./gradlew :app:assembleDebug :app:assembleAndroidTest -DtestBuildType=debug && cd ..",
```
Comment (Kudo) [inline apps/bare-expo/package.json:70]: Would like to explain this more:
`unimodules-test-core` is also subproject of bare-expo.
If we do not specify the target, this will also build `unimodules-test-core:assembleAndroidTest`.
Inside unimodules-test-core, there is a `api 'io.mockk:mockk:1.9.3'` depending on newer android sdk.
The formal solution is to use mockk-android artifact for android instrument test:
```
testImplementation "io.mockk:mockk:{version}"
androidTestImplementation "io.mockk:mockk-android:{version}"
```
See: https://github.com/mockk/mockk/issues/281

However, the dependencies are getting a little hard to be fixed like the formal way.
So the thing is to prevent `unimodules-test-core:assembleAndroidTest` as unimodules-test-core is specific for unit test.

**Comment by Kudo** [comment]
Home CI action is broken in master that #12920 address this.

**Comment by Kudo** [comment]
Feel sorry to request so many people for review. It seems to request automatically from drafted PR to ready review.

File: apps/bare-expo/android/app/src/main/java/dev/expo/payments/MainActivity.java [hunk 2]
```diff
@@ -41,6 +41,19 @@ protected void onCreate(Bundle savedInstanceState) {
         // SplashScreen.show(...) has to be called after super.onCreate(...)
         // Below line is handled by '@expo/configure-splash-screen' command and it's discouraged to modify it manually
         SplashScreen.show(activity, SplashScreenImageResizeMode.COVER, ReactRootView.class, false);
+
+        // Hacky way to prevent onboarding DevMenuActivity breaks detox testing,
+        // which to setup the dev-menu internal setting.
```
Comment (brentvatne) [inline apps/bare-expo/android/app/src/main/java/dev/expo/payments/MainActivity.java:46]:
```suggestion
        // we do this by setting the dev-menu internal setting.
```

**Review by brentvatne — COMMENTED** [review]
this looks good! thanks for doing this in response to manually encountering a regression :)

File: .github/workflows/test-suite.yml [hunk 3]
```diff
@@ -143,3 +143,71 @@ jobs:
           status: ${{ job.status }}
           fields: job,commit,ref,eventName,author,took
           author_name: Test Suite (iOS)
+
+  android:
+    runs-on: macos-latest
```
Comment (EvanBacon) [inline .github/workflows/test-suite.yml:148]: Do we need a macOS for an Android test?

File: apps/bare-expo/android/build.gradle [hunk 4]
```diff
@@ -61,7 +62,13 @@ allprojects {
             url "$rootDir/../node_modules/expo-camera/android/maven"
         }
 
+       maven {
+           // Detox is installed from workspace root npm
+           url "$rootDir/../../../node_modules/detox/Detox-android"
```
Comment (EvanBacon) [inline apps/bare-expo/android/build.gradle:67]: Maybe add detox to the list of symlinks in the app's package.json. This will help prevent a possible future error where detox could have a version mismatch or somn.

File: packages/expo-image/android/build.gradle [hunk 5]
```diff
@@ -34,6 +34,12 @@ buildscript {
 android {
   compileSdkVersion safeExtGet('compileSdkVersion', DEFAULT_COMPILE_SDK_VERSION)
   buildToolsVersion safeExtGet('buildToolsVersion', DEFAULT_BUILD_TOOLS_VERSION)
+
+  compileOptions {
```
Comment (EvanBacon) [inline packages/expo-image/android/build.gradle:38]: Could make sense to move these Java changes to another PR

**Review by EvanBacon — CHANGES_REQUESTED** [review]

File: packages/expo-error-recovery/CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-error-recovery/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-facebook/CHANGELOG.md [hunk 7]
```diff
@@ -16,6 +16,7 @@
 ### 💡 Others
 
 - Removed unnecessary dependency on `unimodules-constants-interface`. ([#12876](https://github.com/expo/expo/pull/12876) by [@tsapeta](https://github.com/tsapeta))
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-facebook/CHANGELOG.md:19]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-file-system/CHANGELOG.md [hunk 8]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated from `unimodules-file-system-interface` to `expo-modules-core`.
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-file-system/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-firebase-analytics/CHANGELOG.md [hunk 9]
```diff
@@ -12,6 +12,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-firebase-analytics/CHANGELOG.md:17]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-firebase-core/CHANGELOG.md [hunk 10]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Removed unnecessary dependency on `unimodules-constants-interface`. ([#12876](https://github.com/expo/expo/pull/12876) by [@tsapeta](https://github.com/tsapeta))
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-firebase-core/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-font/CHANGELOG.md [hunk 11]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated constants interface from `unimodules-constants-interface` to `expo-modules-core`. ([#12876](https://github.com/expo/expo/pull/12876) by [@tsapeta](https://github.com/tsapeta))
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-font/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-gl/CHANGELOG.md [hunk 12]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated from `unimodules-camera-interface`, `unimodules-file-system-interface` to `expo-modules-core`.
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-gl/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-google-sign-in/CHANGELOG.md [hunk 13]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated constants interface from `unimodules-constants-interface` to `expo-modules-core`. ([#12876](https://github.com/expo/expo/pull/12876) by [@tsapeta](https://github.com/tsapeta))
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-google-sign-in/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-haptics/CHANGELOG.md [hunk 14]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-haptics/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-image-manipulator/CHANGELOG.md [hunk 15]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated from `unimodules-file-system-interface` to `expo-modules-core`.
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-image-manipulator/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-in-app-purchases/CHANGELOG.md [hunk 16]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-in-app-purchases/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-intent-launcher/CHANGELOG.md [hunk 17]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-intent-launcher/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-linear-gradient/CHANGELOG.md [hunk 18]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-linear-gradient/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-local-authentication/CHANGELOG.md [hunk 19]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Removed unnecessary dependency on `unimodules-constants-interface`. ([#12876](https://github.com/expo/expo/pull/12876) by [@tsapeta](https://github.com/tsapeta))
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-local-authentication/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-localization/CHANGELOG.md [hunk 20]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-localization/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-mail-composer/CHANGELOG.md [hunk 21]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated from `unimodules-file-system-interface` to `expo-modules-core`.
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-mail-composer/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-module-template/CHANGELOG.md [hunk 22]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-module-template/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-network/CHANGELOG.md [hunk 23]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-network/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-permissions/CHANGELOG.md [hunk 24]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-permissions/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-print/CHANGELOG.md [hunk 25]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated from `unimodules-file-system-interface` to `expo-modules-core`.
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-print/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-screen-capture/CHANGELOG.md [hunk 26]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-screen-capture/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-screen-orientation/CHANGELOG.md [hunk 27]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-screen-orientation/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-secure-store/CHANGELOG.md [hunk 28]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-secure-store/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-sensors/CHANGELOG.md [hunk 29]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated module interfaces from `unimodules-sensors-interface` to `expo-modules-core`. ([#12888](https://github.com/expo/expo/pull/12888) by [@tsapeta](https://github.com/tsapeta))
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-sensors/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-sharing/CHANGELOG.md [hunk 30]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated from `unimodules-file-system-interface` to `expo-modules-core`.
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-sharing/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-sms/CHANGELOG.md [hunk 31]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-sms/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-speech/CHANGELOG.md [hunk 32]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-speech/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-splash-screen/CHANGELOG.md [hunk 33]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-splash-screen/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-sqlite/CHANGELOG.md [hunk 34]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated from `unimodules-file-system-interface` to `expo-modules-core`.
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-sqlite/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-structured-headers/CHANGELOG.md [hunk 35]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-structured-headers/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/expo-video-thumbnails/CHANGELOG.md [hunk 36]
```diff
@@ -13,6 +13,7 @@
 ### 💡 Others
 
 - Migrated from `unimodules-file-system-interface` to `expo-modules-core`.
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/expo-video-thumbnails/CHANGELOG.md:16]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/unimodules-app-loader/CHANGELOG.md [hunk 37]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/unimodules-app-loader/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: packages/unimodules-task-manager-interface/CHANGELOG.md [hunk 38]
```diff
@@ -10,6 +10,10 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
Comment (bbarthec) [inline packages/unimodules-task-manager-interface/CHANGELOG.md:15]:
```suggestion
- Build Android code using Java 8 to fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```

File: apps/bare-expo/android/app/src/main/java/dev/expo/payments/MainActivity.java (hunk 2 above)
Comment (Kudo) [reply apps/bare-expo/android/app/src/main/java/dev/expo/payments/MainActivity.java]: https://github.com/expo/expo/pull/12893/commits/15d31a8393c2b770d83359ce330574f3fd053968

File: apps/bare-expo/android/build.gradle (hunk 4 above)
Comment (Kudo) [reply apps/bare-expo/android/build.gradle]: https://github.com/expo/expo/pull/12893/commits/cc11a20bb22cc962cc9a61645b49c5ec20de26a6

File: packages/expo-image/android/build.gradle (hunk 5 above)
Comment (Kudo) [reply packages/expo-image/android/build.gradle]: makes sense. moving these changes to #12939

File: packages/@unimodules/react-native-adapter/CHANGELOG.md [hunk 39]
```diff
@@ -11,6 +12,10 @@
 
 ### 🐛 Bug fixes
 
+### 💡 Others
+
+- Built with Java 8 to especially fix Android instrumented test build error. ([#12893](https://github.com/expo/expo/pull/12893) by [@kudo](https://github.com/kudo))
```
[the remark this answers is not in the document]
Comment (Kudo) [reply packages/@unimodules/react-native-adapter/CHANGELOG.md]: moving these changes to #12939 and revise the wording accordingly.
going to resolve following comments further. thanks for your comments.

File: .github/workflows/test-suite.yml (hunk 3 above)
Comment (Kudo) [reply .github/workflows/test-suite.yml:152]: That is a good question. Android emulators on Linux seems to be slow on GitHub runners as no hardware acceleration support.
See https://github.com/ReactiveCircus/android-emulator-runner#can-i-use-this-action-on-linux-vms
But it maybe worth to try. Please let me try later tomorrow for CI workers are not busy then.

Comment (Kudo) [reply .github/workflows/test-suite.yml:152]: Some update for ubuntu runner. I've tried a l lot on the past weekend.
The emulator always have the `System UI isn't responding` issue and detox test cannot be installed.
Moreover, it take 15 minutes to boot up the emulator.
I would say android emulator on github ubuntu runner is still not feasible right now.

**Review by brentvatne — APPROVED** [review]

**Review by EvanBacon — APPROVED** [review]
LGTM
